napari_btrack/main.py

```diff
@@ -0,0 +1,220 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from btrack.config import TrackerConfig
+    from magicgui.widgets import Container
+
+    from napari_btrack.config import TrackerConfigs
+
+import logging
+
+import btrack
+import magicgui.widgets
+import napari
+import qtpy.QtWidgets
+from btrack.utils import segmentation_to_objects
+
+import napari_btrack.config
+import napari_btrack.sync
+import napari_btrack.widgets
+
+__all__ = [
+    "create_btrack_widget",
+]
+
+# get the logger instance
+logger = logging.getLogger(__name__)
+
+# if we don't have any handlers, set one up
+if not logger.handlers:
+    # configure stream handler
+    log_fmt = logging.Formatter(
+        "[%(levelname)s][%(asctime)s] %(message)s",
+        datefmt="%Y/%m/%d %I:%M:%S %p",
+    )
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(log_fmt)
+
+    logger.addHandler(console_handler)
+    logger.setLevel(logging.DEBUG)
```

Comment:
I initially had it in the `__init__.py`, but the logger was never set up. I guess this is because when napari loads the plugin, we tell it to import the `napari_btrack.main` module and run the function `create_btrack_widget`. So napari never does `import napari_btrack`, which means the code inside the top level `__init__.py` is never run. But the code to set up the logger will probably be removed anyway when we move the plugin to the `btrack` repository

Reply:
Okay, in that case let's leave it